airflow/providers/apache/beam/operators/beam.py

```diff
@@ -263,12 +263,13 @@ def execute(self, context: 'Context'):
                         process_line_callback=process_line_callback,
                     )
 
-                self.dataflow_hook.wait_for_done(
-                    job_name=dataflow_job_name,
-                    location=self.dataflow_config.location,
-                    job_id=self.dataflow_job_id,
-                    multiple_jobs=False,
-                )
+                if dataflow_job_name and self.dataflow_config.location:
+                    self.dataflow_hook.wait_for_done(
+                        job_name=dataflow_job_name,
+                        location=self.dataflow_config.location,
```

Comment:
assumption is the job_name and self.dataflow_config.location are required parameters